src/devices/bus/a2bus/a2superdrive.cpp

```diff
@@ -0,0 +1,361 @@
+	void devsel_w(uint8_t devsel);
+	void hdsel_w(int hdsel);
+
+
+	required_device<cpu_device> m_65c02;
+	required_region_ptr<u8> m_rom;
+	required_shared_ptr<u8> m_ram;
+
+	required_device<applefdintf_device> m_fdc;
+	required_device_array<floppy_connector, 2> m_floppy;
+
+	floppy_image_device *m_cur_floppy;
+
+	unsigned m_bank_select;
+	unsigned m_side;
+};
+
+
+
+#define SUPERDRIVE_ROM_REGION  "superdrive_rom"
+#define SUPERDRIVE_RAM_TAG "superdrive_ram"
+
+
+
+ROM_START( superdrive )
+	ROM_REGION(0x8000, SUPERDRIVE_ROM_REGION, 0)
+	ROM_LOAD( "341-0438-a.bin", 0x0000, 0x08000, CRC(c73ff25b) SHA1(440c3f84176c7b9f542da0b6ddf4fb13ec326c46) )
+ROM_END
+
+
+const tiny_rom_entry *a2bus_superdrive_device::device_rom_region() const
+{
+	return ROM_NAME( superdrive );
+}
+
+void a2bus_superdrive_device::m65c02_mem(address_map &map)
+{
+	map(0x0000, 0x7fff).ram().share(SUPERDRIVE_RAM_TAG);
+	map(0x0a00, 0x0aff).rw(FUNC(a2bus_superdrive_device::m65c02_r), FUNC(a2bus_superdrive_device::m65c02_w));
+	map(0x8000, 0xffff).rom().region(SUPERDRIVE_ROM_REGION, 0x0000);
+}
+
+
+void a2bus_superdrive_device::device_add_mconfig(machine_config &config)
+{
+
+	M65C02(config, m_65c02,  A2BUS_7M_CLOCK / 3.5);  /* ~ 2.046 MHz */
+	m_65c02->set_addrmap(AS_PROGRAM, &a2bus_superdrive_device::m65c02_mem);
+
+	SWIM1(config, m_fdc, C16M);
+
+	for (auto &floppy: m_floppy)
+		applefdintf_device::add_35_hd(config, floppy);
+
+
+	m_fdc->devsel_cb().set(FUNC(a2bus_superdrive_device::devsel_w));
+	m_fdc->hdsel_cb().set(FUNC(a2bus_superdrive_device::hdsel_w));
+	m_fdc->phases_cb().set(FUNC(a2bus_superdrive_device::phases_w));
+	m_fdc->sel35_cb().set(FUNC(a2bus_superdrive_device::sel35_w));
+}
+
+
+
+a2bus_superdrive_device::a2bus_superdrive_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock) :
+	device_t(mconfig, type, tag, owner, clock),
+	device_a2bus_card_interface(mconfig, *this),
+	m_65c02(*this, "superdrive_65c02"),
+	m_rom(*this, SUPERDRIVE_ROM_REGION),
+	m_ram(*this, SUPERDRIVE_RAM_TAG),
+	m_fdc(*this, "fdc"),
+	m_floppy(*this, "%u", 0U),
+	m_cur_floppy(nullptr),
+	m_bank_select(0),
+	m_side(0)
+{ }
+
+a2bus_superdrive_device::a2bus_superdrive_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock):
+	a2bus_superdrive_device(mconfig, A2BUS_SUPERDRIVE, tag, owner, clock)
+{ }
+
+void a2bus_superdrive_device::device_start()
+{
+	save_item(NAME(m_bank_select));
+	save_item(NAME(m_side));
+}
+
+
+void a2bus_superdrive_device::device_reset()
+{
+	m_bank_select = 0;
+	m_side = 0;
+	m_65c02->reset();
+}
+
+
+// overrides of standard a2bus slot functions
+uint8_t a2bus_superdrive_device::read_c0nx(uint8_t offset)
+{
+	if (machine().side_effects_disabled()) return 0;
+
+	m_bank_select = offset & 0x0f;
+	return 0;
+}
+
+void a2bus_superdrive_device::write_c0nx(uint8_t offset, uint8_t data)
+{
+	if (machine().side_effects_disabled()) return;
+
+	m_bank_select = offset & 0x0f;
+}
```

Comment:
This is probably not a case where suppressing the effects of a write makes sense.  Changing the bank is the *only* effect of the write – it’s not like a situation where it writes a register and also has some other side effect.